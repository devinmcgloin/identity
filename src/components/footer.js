import React from 'react';
import { StaticQuery, graphql } from 'gatsby';

const PureFooter = ({ data }) => {
  const { email, twitter, github, unsplash } = data.social.siteMetadata.social;

  const projects = data.projects.edges.map(e => {
    return {
      title: e.node.frontmatter.title,
      slug: e.node.fields.slug,
    };
  });

  const artwork = data.artwork.edges.map(e => {
    return {
      title: e.node.frontmatter.title,
      slug: e.node.fields.slug,
    };
  });
  return (
    <footer className="flex flex-column flex-row-ns pa4 pa5-l black-70 bt b--black-10 mw8 center">
      <article className="w-100 w-auto-ns mr4-m mr5-l mb4">
        <a
          className="f6 db fw5 pv1 black-70 link dim"
          title="Email"
          href={`mailto:${email}`}
        >
          {email}
        </a>
        <a
          className="f6 db fw5 pv1 black-70 link dim"
          title="Email"
          href={`https://twitter.com/${twitter}`}
        >
          <span className="black-50">twitter</span>/{twitter}
        </a>
        <a
          className="f6 db fw5 pv1 black-70 link dim"
          title="Email"
          href={`https://github.com/${github}`}
        >
          <span className="black-50">github</span>/{github}
        </a>
        <a
          className="f6 db fw5 pv1 black-70 link dim"
          title="Email"
          href={`https://unsplash.com/${unsplash}`}
        >
          <span className="black-50">unsplash</span>/{unsplash}
        </a>
      </article>
      <article className="w-100 w-auto-ns mr4-m mr5-l mb4">
        {projects.map(p => (
          <a
            key={p.slug}
            className="f6 db fw5 pv1 black-70 link dim"
            title={p.title}
            href={p.slug}
          >
            <span className="black-50">project</span>/{p.title}
          </a>
        ))}
      </article>
      <article className="w-100 w-auto-ns mr4-m mr5-l mb4">
        {artwork.map(p => (
          <a
            key={p.slug}
            className="f6 db fw5 pv1 black-70 link dim"
            title={p.title}
            href={p.slug}
          >
            <span className="black-50">interactive</span>/{p.title}
          </a>
        ))}
      </article>
    </footer>
  );
};

const Footer = () => (
  <StaticQuery
    query={graphql`
      {
        social: site {
          siteMetadata {
            social {
              name
              email
              github
              twitter
              unsplash
            }
          }
        }
        artwork: allMdx(
          filter: { fields: { slug: { regex: "/interactive/(.)+/" } } }
          sort: { order: DESC, fields: frontmatter___date }
          limit: 5
        ) {
          edges {
            node {
              frontmatter {
                title
              }
              fields {
                slug
              }
            }
          }
        }
        projects: allMdx(
          filter: { fields: { slug: { regex: "/projects/(.)+/" } } }
          sort: { order: DESC, fields: frontmatter___date }
          limit: 5
        ) {
          edges {
            node {
              frontmatter {
                title
              }
              fields {
                slug
              }
            }
          }
        }
      }
    `}
    render={data => <PureFooter data={data} />}
  />
);
export { PureFooter };
export default Footer;
