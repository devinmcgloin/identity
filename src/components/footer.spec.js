import React from 'react';
import renderer from 'react-test-renderer';
import { PureFooter } from './footer';

describe('Footer', () => {
  const data = {
    site: {
      siteMetadata: {
        social: {
          name: 'Devin McGloin',
          email: 'devin@devinmcgloin.com',
          github: 'devinmcgloin',
          twitter: 'devinmcgloin',
          unsplash: 'devinmcgloin',
        },
      },
    },
    allExperimentsYaml: {
      edges: [
        {
          node: {
            title: 'Solar',
            slug: '/interactive/solar/',
          },
        },
        {
          node: {
            title: "Fermat's Spirals",
            slug: '/interactive/fermat-spiral/',
          },
        },
        {
          node: {
            title: 'Additive Splines',
            slug: '/interactive/additive-splines/',
          },
        },
        {
          node: {
            title: 'Boids',
            slug: '/interactive/boids/',
          },
        },
      ],
    },
    allMdx: {
      edges: [
        {
          node: {
            frontmatter: {
              title: 'Sail',
            },
            fields: {
              slug: '/projects/sail/',
            },
          },
        },
        {
          node: {
            frontmatter: {
              title: 'Probabilistic',
            },
            fields: {
              slug: '/projects/probabilistic/',
            },
          },
        },
        {
          node: {
            frontmatter: {
              title: 'Parking',
            },
            fields: {
              slug: '/projects/parking/',
            },
          },
        },
        {
          node: {
            frontmatter: {
              title: 'clr',
            },
            fields: {
              slug: '/projects/clr/',
            },
          },
        },
        {
          node: {
            frontmatter: {
              title: 'Fokal',
            },
            fields: {
              slug: '/projects/fokal/',
            },
          },
        },
      ],
    },
  };

  it('renders correctly', () => {
    const tree = renderer.create(<PureFooter data={data} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
