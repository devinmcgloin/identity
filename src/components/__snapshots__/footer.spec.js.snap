// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<div
  className="sc-bdVaJa egOjjM"
>
  <footer
    className="flex flex-column flex-row-ns pa4 pa5-l mw8 center"
  >
    <article
      className="w-100 w-auto-ns mr4-m mr5-l mb4"
    >
      <a
        className="f6 db fw5 pv1 black-70 link dim"
        href="mailto:devin@devinmcgloin.com"
        title="Email"
      >
        devin@devinmcgloin.com
      </a>
      <a
        className="f6 db fw5 pv1 black-70 link dim"
        href="https://twitter.com/devinmcgloin"
        title="Email"
      >
        <span
          className="black-50"
        >
          twitter
        </span>
        /
        devinmcgloin
      </a>
      <a
        className="f6 db fw5 pv1 black-70 link dim"
        href="https://github.com/devinmcgloin"
        title="Email"
      >
        <span
          className="black-50"
        >
          github
        </span>
        /
        devinmcgloin
      </a>
      <a
        className="f6 db fw5 pv1 black-70 link dim"
        href="https://unsplash.com/devinmcgloin"
        title="Email"
      >
        <span
          className="black-50"
        >
          unsplash
        </span>
        /
        devinmcgloin
      </a>
    </article>
    <article
      className="w-100 w-auto-ns mr4-m mr5-l mb4"
    >
      <a
        className="f6 db fw5 pv1 black-70 link dim"
        href="/projects/sail/"
        title="Sail"
      >
        <span
          className="black-50"
        >
          project
        </span>
        /
        Sail
      </a>
      <a
        className="f6 db fw5 pv1 black-70 link dim"
        href="/projects/probabilistic/"
        title="Probabilistic"
      >
        <span
          className="black-50"
        >
          project
        </span>
        /
        Probabilistic
      </a>
      <a
        className="f6 db fw5 pv1 black-70 link dim"
        href="/projects/parking/"
        title="Parking"
      >
        <span
          className="black-50"
        >
          project
        </span>
        /
        Parking
      </a>
      <a
        className="f6 db fw5 pv1 black-70 link dim"
        href="/projects/clr/"
        title="clr"
      >
        <span
          className="black-50"
        >
          project
        </span>
        /
        clr
      </a>
      <a
        className="f6 db fw5 pv1 black-70 link dim"
        href="/projects/fokal/"
        title="Fokal"
      >
        <span
          className="black-50"
        >
          project
        </span>
        /
        Fokal
      </a>
    </article>
    <article
      className="w-100 w-auto-ns mr4-m mr5-l mb4"
    >
      <a
        className="f6 db fw5 pv1 black-70 link dim"
        href="/interactive/solar/"
        title="Solar"
      >
        <span
          className="black-50"
        >
          interactive
        </span>
        /
        Solar
      </a>
      <a
        className="f6 db fw5 pv1 black-70 link dim"
        href="/interactive/fermat/"
        title="Fermat's Spirals"
      >
        <span
          className="black-50"
        >
          interactive
        </span>
        /
        Fermat's Spirals
      </a>
      <a
        className="f6 db fw5 pv1 black-70 link dim"
        href="/interactive/additive-splines/"
        title="Additive Splines"
      >
        <span
          className="black-50"
        >
          interactive
        </span>
        /
        Additive Splines
      </a>
      <a
        className="f6 db fw5 pv1 black-70 link dim"
        href="/interactive/boids/"
        title="Boids"
      >
        <span
          className="black-50"
        >
          interactive
        </span>
        /
        Boids
      </a>
    </article>
  </footer>
</div>
`;
